spark3/src/main/java/org/apache/iceberg/spark/procedures/BaseProcedure.java

```diff
@@ -79,10 +79,14 @@ protected Identifier toIdentifier(String identifierAsString, String argName) {
     try {
       catalogAndIdentifier = Spark3Util.catalogAndIdentifier(spark, identifierAsString, tableCatalog);
     } catch (ParseException e) {
-      throw new IllegalArgumentException(String.format("Cannot parse identifier [%s] for argument %s",
+      throw new IllegalArgumentException(String.format("Cannot parse identifier '%s' for argument %s",
           identifierAsString, argName), e);
     }
 
+    Preconditions.checkArgument(catalogAndIdentifier.catalog().equals(tableCatalog), "Cannot run procedure" +
+        " in catalog '%s': Argument %s was set to '%s' which resolves to a table in a different catalog '%s'",
+        tableCatalog.name(), argName, identifierAsString, catalogAndIdentifier.catalog().name());
```

Comment:
This is really wordy. I'd probably simplify it to "Cannot run procedure %s in catalog %s: %s is a table in catalog %s".